.github/workflows/test_package.yml

```diff
@@ -0,0 +1,191 @@
+# branch. If commit is to release branch, the PyPI upload will follow
+# successful tests automatically.
+#
+# Run lint on python source, build step by step the python distro and
+# run all package tests for all OS and python versions
+#
+# If this job succeeds, build step by step all wheels for all OS and python 
+# versions and run the tests in accelerated mode.
+#
+# If the first two jobs succeed, upload the distro and wheels to PyPI
+#
+
+name: test_package
+
+on:
+  # the trigger event for running the action is either a push on master or
+  # release branch
+  
+  push:
+    branches:
+      - master
+      - release
+  
+  # or a pull request to master branch
+  pull_request:
+    branches:
+      - master
+  
+jobs:
+  # This action is split into three jobs:
+  # - Building the distribution linting and testing without acceleration
+  # - Building the wheels for the distribution and testing with acceleration
+  # - Uploading the artifacts to PyPI package if branch is release
+  # The uploading job needs all tests to be finished without error.
+  
+  
+  build_test_dist:
+    # Build the distribution in a matrix. Jobs are done in parallel.
+    # The formal distro which is uploaded to PyPI will be build on
+    # ubuntu-latest for python 3.9, which is the recent one.
+    # As in dist build no compilation takes place, we run all tests
+    # in not accelerated mode.
+    
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 18
+      matrix:
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8, 3.9]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      
+      - name: Setup_Python_${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      
+      # The build test needs numpy and pyflakes. Adding twine enables for
+      # testing and checking the metadata. Adding wheels for package
+      # installation before running the tests
+      
+      - name: install_deps
+        run: pip install numpy pyflakes
+      
+      - name: build_sdist
+        run: python setup.py sdist
+      
+      - name: check_metadata
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.9'
+        run: |
+          pip install twine
+          twine check dist/*
+
+      - name: upload_sdist
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.9'
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: dist/*.tar.gz
+          
+      - name: copy_file
+        run: mv dist/sgp4*.* dist/sgp4.tar.gz
+
+      - name: install_dist
+        run: |
+          pip install dist/sgp4.tar.gz
+          python -c "from sgp4.api import accelerated; print(accelerated)"
+
+      - name: run_tests
+        run: python -m sgp4.tests
+
+
+  build_test_wheels:
```

Comment:
Could you remind me of the reason that this does not have the same `if` clause that `upload_to_pypi` does? I think I would be happy with normal PR's and commits just running tests against Python versions, if that could skip building the wheels.

Reply:
See above: actually python testing does only test the unaccelerated version. As I wanted to test the accelerated version as well, I build the wheels and test in the cibuildwheel docker. To be decided about the strategy.